checkov/terraform/plan_parser.py

```diff
@@ -165,6 +166,10 @@ def _prepare_resource_block(
             resource_conf[TF_PLAN_RESOURCE_CHANGE_ACTIONS] = changes.get("change", {}).get("actions") or []
             resource_conf[TF_PLAN_RESOURCE_CHANGE_KEYS] = changes.get(TF_PLAN_RESOURCE_CHANGE_KEYS) or []
 
+        provisioners = conf.get(TF_PLAN_RESOURCE_PROVISIONERS) if conf else None
+        if provisioners:
+            resource_conf[TF_PLAN_RESOURCE_PROVISIONERS] = provisioners
```

Comment:
I think the problem with this approach is, that the `provisioners` in HCL looks different than in the plan file. So, a custom policy would need to cover both cases.

Reply:
@gruebel good callout. I've updated it to match as best I could, but note that Terraform transforms the command and removes some of it, so it won't match completely. Now this policy:

```yaml
metadata:
 name: "Test"
 id: "CKV2_CUSTOM_1"
 guidelines: "Policy description"
 category: "GENERAL_SECURITY"
scope:
  provider: "aws"
definition:
  cond_type: attribute
  resource_types:
    - aws_instance
  attribute: "provisioner/local-exec"
  operator: not_exists
```

Works for both TF and Plan files using all methods documented here: https://developer.hashicorp.com/terraform/language/resources/provisioners/syntax

![Screenshot 2023-10-19 at 1 23 08 AM](https://github.com/bridgecrewio/checkov/assets/28880387/88071d2e-13b4-4dfe-acc1-12cc65e8754a)
![Screenshot 2023-10-19 at 1 22 32 AM](https://github.com/bridgecrewio/checkov/assets/28880387/80a4eb33-11de-4cdb-bb89-8297942c48fb)

Reply:
sounds good